openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TagRepository.java

```diff
@@ -109,6 +122,82 @@ public void setFullyQualifiedName(Tag tag) {
     }
   }
 
+  public BulkOperationResult bulkAddAndValidateTagToAssets(
```

Comment:
is this async operation?

Reply:
@harshach , this is sync operation , similar to the current implementation that we have for adding assets to glossaryTerm on Assets tab

<img width="1116" alt="image" src="https://github.com/user-attachments/assets/67b5e21d-4ddb-4e81-93cc-ef36a1b5a069">